src/protocol/prf_sharding/mod.rs

```diff
@@ -940,4 +993,119 @@ pub mod tests {
+            let world = TestWorld::default();
+
+            let records: Vec<PreAggregationTestOutput> = vec![
+                test_output::<Gf5Bit, Gf3Bit>(17, 7),
+                test_output::<Gf5Bit, Gf3Bit>(20, 0),
+                test_output::<Gf5Bit, Gf3Bit>(20, 3),
+                test_output::<Gf5Bit, Gf3Bit>(12, 5),
+                test_output::<Gf5Bit, Gf3Bit>(20, 7),
+                test_output::<Gf5Bit, Gf3Bit>(18, 0),
+                test_output::<Gf5Bit, Gf3Bit>(12, 0),
+                test_output::<Gf5Bit, Gf3Bit>(12, 7),
+                test_output::<Gf5Bit, Gf3Bit>(12, 7),
+                test_output::<Gf5Bit, Gf3Bit>(12, 7),
+                test_output::<Gf5Bit, Gf3Bit>(12, 4),
+            ];
+
+            let mut expected = [0_u128; 32];
+            expected[12] = 30;
+            expected[17] = 7;
+            expected[20] = 10;
+
+            let result: Vec<_> = world
+                .semi_honest(records.into_iter(), |ctx, input_rows| async move {
+                    let validator = ctx.validator();
+                    let ctx = validator.context();
+                    do_aggregation::<_, Gf5Bit, Gf3Bit, Fp32BitPrime, _>(ctx, input_rows)
+                        .await
+                        .unwrap()
+                })
+                .await
+                .reconstruct();
+            assert_eq!(result, &expected);
+        });
+    }
+
+    #[test]
+    fn semi_honest_move_value_to_single_bucket() {
+        run(|| async move {
+            let world = TestWorld::default();
+            let mut rng: rand::rngs::ThreadRng = thread_rng();
+
+            let breakdown_key = rng.gen_range(0..32);
+
+            let value = Fp32BitPrime::truncate_from(10_u128);
+            let mut expected = [Fp32BitPrime::truncate_from(0_u128); 32];
+            expected[breakdown_key] = value;
+
+            let result: Vec<_> = world
+                .semi_honest(
+                    (
+                        get_bits::<Fp32BitPrime>(breakdown_key.try_into().unwrap(), Gf5Bit::BITS),
+                        value,
+                    ),
+                    |ctx, (breakdown_key_share, value_share)| async move {
+                        let validator = ctx.validator();
+                        let ctx = validator.context();
+                        move_single_value_to_bucket::<Gf5Bit, _, _, Fp32BitPrime>(
+                            ctx.set_total_records(1),
+                            RecordId::from(0),
+                            breakdown_key_share,
+                            value_share,
+                            None,
+                        )
+                        .await
+                        .unwrap()
+                    },
+                )
+                .await
+                .reconstruct();
+            assert_eq!(result, &expected);
+        });
+    }
+
+    #[test]
+    fn semi_honest_move_value_to_single_bucket_not_power_two() {
+        const BREAKDOWN_COUNT: usize = 50;
```

Comment:
There is too much duplication between `semi_honest_move_value_to_single_bucket` and `semi_honest_move_value_to_single_bucket_not_power_two`.

I recommend just merging them into a single test that loops many times, trying many combinations of "num_breakdowns" (some of which are powers of two, and some of which are not), and value.

We do not have good test coverage as it is, since you're only testing 50, and only for a single value.